lib/Applify.pm

```diff
@@ -310,30 +319,32 @@ sub app {
 
 sub _calculate_option_spec {
   my ($self, $option) = @_;
-  my $spec = $self->_attr_to_option($option->{name});
+  my @spec = $self->_attr_to_option($option->{name});
 
   if (ref $option->{alias} eq 'ARRAY') {
-    $spec .= join '|', '', @{$option->{alias}};
+    $spec[0] .= join '|', '', @{$option->{alias}};
   }
 
-  if    ($option->{type} =~ /^(?:bool|flag)/i) { $spec .= '!' }
-  elsif ($option->{type} =~ /^inc/)            { $spec .= '+' }
-  elsif ($option->{type} =~ /^str/)            { $spec .= '=s' }
-  elsif ($option->{type} =~ /^int/i)           { $spec .= '=i' }
-  elsif ($option->{type} =~ /^num/i)           { $spec .= '=f' }
-  elsif ($option->{type} =~ /^file/)           { $spec .= '=s' }    # TODO
-  elsif ($option->{type} =~ /^dir/)            { $spec .= '=s' }    # TODO
-  else                                         { die 'Usage: option {bool|flag|inc|str|int|num|file|dir} ...' }
+  if    ($option->{type} =~ /^(?:bool|flag)/i) { $spec[0] .= '!' }
+  elsif ($option->{type} =~ /^inc/)            { $spec[0] .= '+' }
+  elsif ($option->{type} =~ /^str/)            { $spec[0] .= '=s' }
+  elsif ($option->{type} =~ /^int/i)           { $spec[0] .= '=i' }
+  elsif ($option->{type} =~ /^num/i)           { $spec[0] .= '=f' }
+  elsif ($option->{type} =~ /^(file|dir)/) {
+    $spec[0] .= '=s';
+    $spec[1] = sub { return Mojo::File->new($_[1]) }
```

Comment:
Instead of forcing users to download Mojolicious, I think I would rather have an attribute that told the module which "file" module to use. It could even have a default, where it checks if it can load `qw(Mojo::File Path::Tiny Path::Class)` (Not sure about what should be in the list though)